### ISSUE TITLE
Add tool linking to dbSNP and ClinVar r.e. https://github.com/openvar/VVweb/issues/32

### ISSUE THREAD

**Issue by PeteCausey-Freeman** [openvar/variantValidator#210]
We decided to add a module to VV to perform this task. Iteration 1 will use the SPDI APIs which we will replace when we build a Python SPDI tool.

**Comment by bev-t** [comment]
The module fails for variant NC_000006.11:g.102503229C>T. It successfully converts the variant to the SPDI NC_000006.11:102503228:C:T but then doesn't find an RSID. Upon checking, there is no RefSeq for this variant. 
Rather than throwing an exception for this scenario, should we treat as a successful call but return an empty dictionary and tell VV to ignore it when attempting to create the URL link? What do you think?

**Comment by PeteCausey-Freeman** [comment]
Agreed. It isn't an exception, and should be a blank RSID. Good idea

**Comment by bev-t** [comment]
Made following changes to code and pushed to BEVERLEY_TAYLOR branch.

1. **test_variant_external_resources**: Added 2 additional unit tests for 1. when no RSID found and 2. test URLs are returned for a valid variant.
2. **variant_external_resources**: amended the code so that it returns an empty dictionary for a 404 not found error.
3. updated README.txt with latest changes

**Comment by PeteCausey-Freeman** [comment]
Hi @bev-t . I have come back from a stint of grant writing and holiday.

So catching up. Did you manage to install VariantValidator locally.

**Comment by bev-t** [comment]
Hi @PeteCausey-Freeman, I don't have the instructions, can you send them to me please or point me in the right direction? I can start on it today.

**Comment by bev-t** [comment]
Hi @PeteCausey-Freeman, I have vv installed locally now.

**Comment by PeteCausey-Freeman** [comment]
Hi @bev-t 

Nice work. OK. Next step is to install https://github.com/openvar/VVweb. Don't worry about celery rabbit or any of this stuff. We just need it running on your system .See the instructions. 

Ultimately we need the command python manage.py runserver to run the vv website locally. You can then see changes you make such as adding your web updates.

**Comment by bev-t** [comment]
Hi @PeteCausey-Freeman,

I'm setting up the SECRET_KEY in local_settings.py file for VVweb. Is this something I choose or do I need to get it from somewhere?

Thanks
Bev

**Comment by PeteCausey-Freeman** [comment]
I believe, remembering off the top of my head you can find an online secret key generor

**Comment by bev-t** [comment]
Found it 👍

**Comment by bev-t** [comment]
Hi @PeteCausey-Freeman, 

I'm getting this error when running variant_validator.py - I’ve tried removing and reinstalling the environment vvenv and still get the exact same error (also get same error in vvweb manage.py makemigrations).
  

File "/home/bevt/anaconda3/envs/vvenv/lib/python3.6/site-packages/vvhgvs/dataproviders/uta.py", line 517, in _ensure_schema_exists 

    self.url.schema, self.url)) 

vvhgvs.exceptions.HGVSDataNotAvailableError: specified schema (uta_20180821) does not exist (url=postgresql://vvadmin:vvadmin@localhost/myproject/uta_20180821) 

  

The relevant section in the default.ini file is: 

[postgres] 

host = localhost 

database = uta 

version = uta_20180821 

user = uta_admin 

 password removed

I can't see where the script is getting the values from to construct the uta URL – I.e. it should be user uta_admin. Any ideas please? 

Thanks
Bev

**Comment by PeteCausey-Freeman** [comment]
Hi Bev,

Perfect, this is why I wanted you to go through the installation to highlight errors.

OK, Database setup is here.

https://github.com/openvar/variantValidator/blob/master/docs/INSTALLATION.md

Please feel free to edit the installation instructions and push up so I can merge. This is now your project too, so feel free to propose changes.

Welcome to the team :)

**Comment by bev-t** [comment]
Thanks :-)
I've already followed 'Setting up UTA database (Optional, PostGreSQL >=9.5)' and verified that it is installed. It's the python script variant_validator.py that is picking up the wrong user and location for UTA when it's constructing the URL. I can't trace where it is getting the values from - thought it was default.ini but that doesn't see to be the case. Does it create any environment variables anywhere?

Thanks
Bev

**Comment by bev-t** [comment]
Hi @PeteCausey-Freeman ,

I've resolved the issue temporarily by editing the uta.py file and hard coding the uta connection arguments:

conn_args = dict(
            host="localhost", #self.url.hostname,
            port="5432", #self.url.port,
            database="uta", #self.url.database,
            user="uta_admin", #self.url.username,
            password="_removed_", #self.url.password,

so it is working now and I get a result when I run variant_validator.py. Next I'll make the same changes to uta.py in vvweb and see how far I can get with it.
Thanks
Bev

**Comment by PeteCausey-Freeman** [comment]
Hiya

Sorry, been on meetings

See https://github.com/openvar/variantValidator/blob/master/docs/MANUAL.md under configuration. does this help?

**Comment by bev-t** [comment]
Hi @PeteCausey-Freeman,

Finally! All installed yeaaaa :-). I had to do a bit of hard-coding and temporarily change some settings to enable the makemigrations, migrate and runserver scripts to run, so I could do with trying to work out what went wrong to be able to update the installation instructions before I move on. I may need to put a quick Zoom call in if I can't resolve by myself.

![image](https://user-images.githubusercontent.com/57769482/92247813-a7d2dd80-eebf-11ea-8d85-006af8f861a4.png)

Thanks
Bev

**Comment by PeteCausey-Freeman** [comment]
Sure. Email me directly and we can set up.

**Comment by bev-t** [comment]
Hi @PeteCausey-Freeman,

**VariantValidator**
I've added variant_external_resources.py and the amended INSTALLATION.md to the **develop_v3** branch of VariantValidator for you to review. 

**vvweb**
I've made changes locally to add the dbSNP and clinvar links to vvweb and all is working well. The new module variant_external_resources.py will need adding to the vvweb installation. My understanding is that during installation, vvweb takes the latest version of VariantValidator from master. So do you want me to wait until you've reviewed the changes in VariantValidator and merged with master so variant_external_resources.py will be available to use in vvweb? or should I checkout a new branch in vvweb and manually add the new module?


Thanks
Bev

**Comment by bev-t** [comment]
I’ve pushed the following changes to git: 

**VariantValidator** 

- Branch [typoon210] - I spotted a typo on the INSTALLATION.md 

- Branch [pytestsfor210] Refactored the error handling for variant_external_resources.py so it returns a dictionary instead of a system exit if an error occurs. Added test_variant_external_resources with pytests and added README.md with info on the tests I’ve either implemented or suggested but not done right now. 

**VVWeb** 

- Branch [vvweb210] Integrated variant_external_resources into the API. 

Just to note, the tool doesn't return links for variant descriptions that use a ‘chr’ location e.g. chr17:50198002C>A because NCBI Variant Services requires the full GenColl assembly accession number e.g. GCA_000002285.2. This could be added as an enhancement. 

Bev
